content/articles/understanding-server-side-request-forgery-attacks/index.md

```diff
@@ -0,0 +1,121 @@
+
+SSRF attacks are most common to web applications which fetch data from a server to display it to the client. When the application fetches the resources without validating the supplied URL, the application becomes vulnerable to SSRF. An attacker can alter the application’s request and send a malicious request.
+
+### Prerequisites
+To follow along with this tutorial, some knowledge about intercepting traffic flow using [Burpsuite](https://portswigger.net/burp/communitydownload), [OWASP Zap](https://www.zaproxy.org/download/), or any other tool is neccesary.
+
+### Overview of SSRF attacks.
+Server-Side Request Forgery (SSRF) is a web application vulnerability that occurs when a server-side application is induced to make arbitrary HTTP requests to an arbitrary domain chosen by the attacker. This simply means that the web application fetches remote resources without validating the user-supplied URL.
+
+Normally, when an attacker tries to access resources a server from another source other than the "known" web application, they will mostly get blocked. But in our case (SSRF), the server will receive the request as coming from the web application and hence the request will be accepted.
+
+The limiting power to the attacker's request will be the permission range given to this web application.
+
+SSRF attacks can take place in different ways, such as:
+- *Against the hosting server* - In this, the requests are sent to the web server hosting the web application. The attacker can also leverage this attack to perform another kind of attack.
+- *Against other back-end systems* - The attacker can leverage the attack above to access other services at the back-end of the same network as that of the web server hosting the application. These other servers might not block requests since they are originating from a known source.
+
+### Types of SSRF attacks.
+Some of the common types of SSRF attacks are:
+- Regular SSRF
+- Blind SSRF
+
+#### 1. Regular SSRF
+In this type of attack, the attacker can receive feedback. For example, if the attacker requested files, they can view them once they complete the request.
+
+#### 2. Blind SSRF
+In this type of attack, the attacker cannot get a response from the server either via an error message or an HTTP response.
+
+Although the attacker doesn’t get a response, it doesn’t mean that this attack will not be valuable to the attacker.
+
+### Testing for SSRF attacks.
+We can test our applications for SSRF vulnerabilities in a couple of ways.
+
+The most recommended one is manual code review to check whether the URL inputs are being validated.
+
+Sometimes, we might find that we don’t have access to the source code of our applications, but this should not hinder us from testing our applications.
+
+To show how we can test for SSRF in cases where you don't have access to the source code, I will use Labs from [Portswigger](https://portswigger.net/web-security/ssrf). We will look at a lab for a local server.
+
```

Comment:
@Dyrstiu Add the link to the lab in use for your demonstrations.

Reply:
@Dyrstiu

Reply:
The link attached is for the labs by portswigger

Reply:
The link points to a generic SSRF page. I'm talking about the lab itself.

https://portswigger.net/web-security/ssrf/lab-basic-ssrf-against-localhost

Reply:
@geoffrey45 I have noticed the error and made the corrections .thank you